plat/drivers/virtio/virtio_net.c

```diff
@@ -997,7 +1103,7 @@ static int virtio_netdev_rxtx_alloc(struct virtio_net_device *vndev,
 	 * ...
 	 * Virtqueue-ctrlq
 	 */
-	for (i = 0; i < vndev->max_vqueue_pairs; i++) {
+	for (i = 0; i < total_vqs; i++) {
```

Comment:
Shouldn't this be `vq_pairs`?

Reply:
No, it is not. vq_pairs are half of the total virtqueues supported by the device. Hence while initializing, we have to initialize all of the virtqueues.

Reply:
This is definitely wrong and using `total_vqs` will cause an out-of-bounds access in the following lines: for RX at `vndev->rxqs[i]`, where `rxqs` only has `conf->nb_rx_queues` (< `total_vqs`!) elements.